hack/update-taps.sh

```diff
@@ -0,0 +1,176 @@
+#! /usr/bin/env bash
+
+# Copyright 2024 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+
+
+# Retrieve latest version from given Knative repository tags
+# On 'main' branch the latest released version is returned
+# On 'release-x.y' branch the latest patch version for 'x.y.*' is returned
+# Similar to hack/library.sh get_latest_knative_yaml_source()
+function get_latest_release_version() {
+    local org_name="$1"
+    local repo_name="$2"
+    local major_minor=""
+    local version
+    version="$(git ls-remote --tags --ref https://github.com/"${org_name}"/"${repo_name}".git \
+      | grep "${major_minor}" \
+      | cut -d '-' -f2 \
+      | cut -d 'v' -f2 \
+      | sort -Vr \
+      | head -n 1)"
+    echo "${version}"
+}
+
+# Return the major version of a release.
+# For example, "v0.2.1" returns "0"
+# Parameters: $1 - release version label.
+function major_version() {
+  local release="${1//v/}"
+  local tokens=(${release//\./ })
+  echo "${tokens[0]}"
+}
+
+# Return the minor version of a release.
+# For example, "v0.2.1" returns "2"
+# Parameters: $1 - release version label.
+function minor_version() {
+  local tokens=(${1//\./ })
+  echo "${tokens[1]}"
+}
+
+# Return the release build number of a release.
+# For example, "v0.2.1" returns "1".
+# Parameters: $1 - release version label.
+function patch_version() {
+  local tokens=(${1//\./ })
+  echo "${tokens[2]}"
+}
+
+# Generate file based on the inlined HEREDOC template.
+function generate_tap_file() {
+  out=$1
```

Comment:
you could mark all those variables as `local`

Reply:
Well, I had that though, but then shellcheck wanted me to split declaration and I was too lazy to do it. :)

Reply:
I'll change it NP.

Reply:
https://github.com/knative/homebrew-client/pull/138/commits/24dfe8e67fedc8ba03dc697f97ed1a520669dded

Reply:
strange, wonder what the reasoning of shellcheck is for insisting of having declaration and initialization in two statements and why it doesn't complain about global variables ;-)